server/src/domain/game/User.test.ts

```diff
@@ -0,0 +1,27 @@
+import { describe, expect, it, vi } from 'vitest';
+
+import { User } from './User';
+
+vi.mock('./CharacterRepositoryPostgres', () => {
```

Comment:
Mocking CharacterRepository here is the right move, but we shouldn't rely on `CharacterRepositoryPostgres instead use` abstracted `ICharacterRepository` interface. So I think we should rewrite it to this:

```ts
export class User {
    readonly id: UUID = randomUUID();
    profileIMG: string = '';

    constructor(characterRepository: ICharacterRepository) {} // Add constructor to user which accepts any repository that implements ICharacterRepository interface
}
```
and then pass our fake repository in constructor like this:

```ts
const fakeRepo: ICharacterRepository = {
    fetchCharacters: vi.fn().mockReturnValue([
        { id: 'character1', userId: 'user1', roomId: 'room1' },
        { id: 'character2', userId: 'user1', roomId: 'room2' }
    ])
};

const user = new User(fakeRepo);

// And the rest stays the same

```

Reply:
Oh yes this makes so much more sense, will do.